textBasedAdventure.js

```diff
@@ -1,5 +1,203 @@
 const readline = require('readline-sync')
+const readlineSync = require('readline-sync')
+// const currentDefaultOptions = readlineSync.setDefaultOptions([newDefaultOptions])
 
-let nameInput = readline.question("Enter your name: ")
+let playGame = true
+let playerHealth = 0
+let playerAttack = 0
+let playerDefense = 0
+let playableRace = ["Elf","Human","Orc"]
+let playerRaceStat = {
+    "Elf":[12,8,8], //array = healh,attack,Defence
+    "Human":[10,10,10],
+    "Orc":[8,12,12]
+}
+let availWeapon =["Sword","Axe","Spear","Fist"]
+let weaponStat = {
+    Sword: 2,
+    Axe: 3,
+    Spear: 4,
+    Fist: 6
+}
+
+let floorEnemy = [
+    {race : "Gaint Rat", stat :[6,4,4]},
+    {race : "orc", stat :[10,6,6]},
+    {race : "skeleton", stat :[4,4,4]},
+    {race : "goblin", stat :[8,8,8]}
+]
+let floorBoss = [
+    { boss : "Dragon", stat:[30,12,12]},
+    { boss : "Lich", stat :[30,13,13]},
+    { boss: "Dark Knight", stat :[30,11,11]},
+]
+let roundOneCounter = 0
+let position = {
+    x : 0,
+    y : 0
+};
+
+const healthCheck = () =>{
+    if(playerHealth <= 0){
+        return playGame = false; 
+    }
+}
+
+
+//choose which direction to go
+const upDownLeftRight = () =>{ 
+    let upDownLeftRightChoice;
+    while(upDownLeftRightChoice !== "u" && upDownLeftRightChoice !== "l" && upDownLeftRightChoice !== "d" && upDownLeftRightChoice !== "r" ){
+        upDownLeftRightChoice = readline.question('Which room would you like to move to? Up [u], Down [d], left [l], right [r]');
+        console.log(upDownLeftRightChoice)
+    }
+
+    if(String(upDownLeftRightChoice) === "u"){
+       return position["y"] += 1;
+    } else if(String(upDownLeftRightChoice) === "l"){
+       return position["x"] -= 1;
+    } else if(String(upDownLeftRightChoice) === "d"){
+       return position["y"] -= 1;
+    } else if(String(upDownLeftRightChoice) === "r"){
+       return position["x"] += 1;
+}
+}
+
+//random number generator to determine if chest found
+const chestChance = () =>{
+    let cContains;
+    cContains = Math.floor((Math.random() * 4) + 1);
```

Comment:
Nothing wrong about this code, but you could have just returned Math.floor((Math.random() * 4) + 1);